internal/consensus/pbts_test.go

```diff
@@ -1,56 +1,299 @@
 package consensus
 
 import (
+	"bytes"
+	"context"
 	"testing"
 	"time"
 
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 
+	"github.com/tendermint/tendermint/abci/example/kvstore"
+	tmpubsub "github.com/tendermint/tendermint/libs/pubsub"
 	tmtimemocks "github.com/tendermint/tendermint/libs/time/mocks"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
 	"github.com/tendermint/tendermint/types"
 )
 
+type pbtsTestHarness struct {
+	s                 pbtsTestConfiguration
```

Comment:
Consider maybe capturing the `*testing.T` in this struct at construction time, so that you don't have to remember to plumb it in through all the methods? (The methods would still need to reference the field, but the caller wouldn't)

Reply:
Separately: Consider calling this field `config` or something? I can't see a clear mnemonic to `s`, though I applaud the brevity 🙂

Reply:
Alternatively: Embed the config type, so you can use its members as if they were fields of the harness?

Reply:
made embedded, held onto the testing.T